File: Laba1/Laba1/WaterTap.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Laba1
{
    class WaterTap
    {
        public bool State;

        internal void Wash_pig(pig obj)
        {
            if (State)
            {
                obj.Dirty = 0;
            }
        }
        internal void Wash_pepper(GorikiyPepper obj)
        {
            if (State)
            {
                obj.Dirty = 0;
            }
        }
        internal void Wash_petrushka(petrushka obj)
        {
            if (State)
            {
                obj.Dirty = 0;
            }
        }
    }
}

File: Laba1/src/main/WaterTap.java
package main;

public class WaterTap {

    private boolean state;

    public boolean isState() {
        return state;
    }

    public void setState(boolean state) {
        this.state = state;
    }

    public void WashPig(Pig obj){
        if(state){
            obj.setDirty(0);
        }
    }
    public void WahPepper(GorikiyPepper obj){
        if(state){
        	obj.setDirty(false);
        }
    }

    public void WashPetrushka(Petrushka obj){
        if(state){
        	obj.setDirty(0);
        }
    }
}
